pkg/yurtmanager/controller/yurtappoverrider/yurtappoverrider_controller.go

```diff
@@ -0,0 +1,223 @@
+	scheme        *runtime.Scheme
+	recorder      record.EventRecorder
+	Configuration config.YurtAppOverriderControllerConfiguration
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(c *appconfig.CompletedConfig, mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileYurtAppOverrider{
+		Client:        mgr.GetClient(),
+		scheme:        mgr.GetScheme(),
+		recorder:      mgr.GetEventRecorderFor(ControllerName),
+		Configuration: c.ComponentConfig.YurtAppOverriderController,
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(ControllerName, mgr, controller.Options{
+		Reconciler: r, MaxConcurrentReconciles: concurrentReconciles,
+	})
+	if err != nil {
+		return err
+	}
+
+	yurtappoverriderPredicates := predicate.Funcs{
+		CreateFunc: func(evt event.CreateEvent) bool {
+			obj, ok := evt.Object.(*appsv1alpha1.YurtAppOverrider)
+			if !ok {
+				return ok
+			}
+			if err := r.(*ReconcileYurtAppOverrider).updatePools(obj); err != nil {
+				klog.Errorf("fail to update deployments belonging to obj: %v", err)
+			}
+			return true
+		},
+		DeleteFunc: func(evt event.DeleteEvent) bool {
+			obj, ok := evt.Object.(*appsv1alpha1.YurtAppOverrider)
+			if !ok {
+				return ok
+			}
+			if err := r.(*ReconcileYurtAppOverrider).updatePools(obj); err != nil {
+				klog.Errorf("fail to update deployments belonging to obj: %v", err)
+			}
+			return true
+		},
+		UpdateFunc: func(evt event.UpdateEvent) bool {
+			obj, ok := evt.ObjectOld.(*appsv1alpha1.YurtAppOverrider)
+			if !ok {
+				return ok
+			}
+			if err := r.(*ReconcileYurtAppOverrider).updatePools(obj); err != nil {
+				klog.Errorf("fail to update deployments belonging to obj: %v", err)
+			}
+			return true
+		},
+		GenericFunc: func(evt event.GenericEvent) bool {
+			return false
+		},
+	}
+
+	// Watch for changes to YurtAppOverrider
+	err = c.Watch(&source.Kind{Type: &appsv1alpha1.YurtAppOverrider{}}, &handler.EnqueueRequestForObject{}, yurtappoverriderPredicates)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// +kubebuilder:rbac:groups=apps.openyurt.io,resources=yurtappoverriders,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=deployments/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=core,resources=events,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=controllerrevisions,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile reads that state of the cluster for a YurtAppOverrider object and makes changes based on the state read
+// and what is in the YurtAppOverrider.Spec
+func (r *ReconcileYurtAppOverrider) Reconcile(_ context.Context, request reconcile.Request) (reconcile.Result, error) {
+
```

Comment:
please update deployments that owner reference is YurtAppOverrider specified by request, instead of updating nodepools that maybe cause all deployments will be updated.